keras_nlp/models/bert/bert_tasks_test.py

```diff
@@ -34,82 +36,82 @@ def setUp(self):
             max_sequence_length=128,
             name="encoder",
         )
-        self.classifier = BertClassifier(self.backbone, 4, name="classifier")
-        self.batch_size = 8
-        self.input_batch = {
-            "token_ids": tf.ones(
-                (self.batch_size, self.backbone.max_sequence_length),
-                dtype="int32",
-            ),
-            "segment_ids": tf.ones(
-                (self.batch_size, self.backbone.max_sequence_length),
-                dtype="int32",
-            ),
-            "padding_mask": tf.ones(
-                (self.batch_size, self.backbone.max_sequence_length),
-                dtype="int32",
-            ),
-        }
+        self.vocab = ["[PAD]", "[UNK]", "[CLS]", "[SEP]", "[MASK]"]
+        self.vocab += ["the", "quick", "brown", "fox", "."]
+        self.preprocessor = BertPreprocessor(
+            BertTokenizer(vocabulary=self.vocab),
+            sequence_length=8,
+        )
+        self.classifier = BertClassifier(
+            self.backbone,
+            4,
+            preprocessor=self.preprocessor,
+        )
+        self.classifier_no_preprocessing = BertClassifier(
+            self.backbone,
+            4,
+            preprocessor=None,
+        )
 
-        self.input_dataset = tf.data.Dataset.from_tensor_slices(
-            self.input_batch
+        self.raw_batch = tf.constant(
+            [
+                "the quick brown fox.",
+                "the slow brown fox.",
+                "the smelly brown fox.",
+                "the old brown fox.",
+            ]
+        )
+        self.preprocessed_batch = self.preprocessor(self.raw_batch)
+        self.raw_dataset = tf.data.Dataset.from_tensor_slices(
```

Comment:
Do we have enough testing (either here or in `pipeline_model_test.py`) for non-`Dataset` inputs? A user browsing this code could be forgiven for thinking `Dataset` was all we supported.

Reply:
My suspicion seems to be correct. Both the following calls throw in this branch:
```python
classifier = keras_nlp.models.BertClassifier.from_preset("bert_tiny_uncased_en")
classifier("I love modular workflows in keras-nlp!")
classifier(["I love modular workflows in keras-nlp!", "I hate spaghetti code."])
```

The only thing I could get to work outside `tf.data.Dataset` was a tensor input:
```python
classifier(tf.constant("I love modular workflows in keras-nlp!", shape=(1,))
```

Reply:
There is a good amount of testing for non-dataset inputs (especially for the pipeline utility itself, which is why that test file is fairly bulky). We could add more testing for raw inputs like this. But I think what you are hitting here is a few bugs that we need to clean up.
 - For the first call you show, I would expect that to hit our issue where our models will give a confusing error for unbatched inputs. That is https://github.com/keras-team/keras-nlp/issues/395
 - For the second call, that is an issue where that tuple is is interpreted as representing multiple segments, not multiple samples. So that call two is unbatched. That is https://github.com/keras-team/keras-nlp/issues/396
 - The third call you show is the only one that will result in a valid batch dimension, so the only one that does not error out here.

Clearly the UX is bad there. I can open some PRs with proposed fixes for both of those today. But I think the behavior you are describing is correct for the current state of things on master.

Reply:
Just FYI, today, if you wanted to pass a raw string. You would need to have a call like `classifier([["I love modular workflows in keras-nlp!"]])` (though I'm not advocating we keep that behavior!). First list is batch dim, second list is segment enumeration.

Reply:
Thanks for the clarification @mattdangerw! At a minimum we shouldn't be throwing. Good communication with the user will be a whole other level of challenge.

Reply:
I would actually love for every usage you showed to "just work," as I think that would be the most intuitive experience there. But can follow up on PRs for those respective issues.